XL: latest master error with 500 internal server error.

```
Goroutine: 3
--- FAIL: TestRemovePartiallyUploaded (38.18s)
        api-functional-v4_test.go:302: Error: 500 Internal Server Error
FAIL

Goroutine: 131
PASS

Goroutine: 8
PASS

Goroutine: 252
--- FAIL: TestRemovePartiallyUploaded (28.65s)
        api-functional-v4_test.go:298: Error: 500 Internal Server Error
FAIL

Goroutine: 896
--- FAIL: TestRemovePartiallyUploaded (38.86s)
        api-functional-v4_test.go:298: Error: 500 Internal Server Error
FAIL
```

For 1000 go-routine run 

```
func main() {
        var wg sync.WaitGroup
        f, _ := os.Create("output.log")
        defer f.Close()
        testCmd := "./minio.test -test.timeout 3600s"
        if len(os.Args[1]) != 0 {
                testCmd = fmt.Sprintf("./minio.test -test.timeout 3600s -test.run %s", os.Args[1])
        }
        for i := 0; i < 1000; i++ {
                wg.Add(1)
                go func(routineId int) {
                        defer wg.Done()

                        out, err := exec.Command("sh", "-c", testCmd).Output()
                        if err != nil {
                                fmt.Println(err.Error())
                        }
                        fmt.Fprintf(f, "\nGoroutine: %d\n", routineId+1)
                        fmt.Fprintf(f, string(out))

                }(i)
        }
        wg.Wait()

}
```

--- a/server-rlimit-nix.go
+++ b/server-rlimit-nix.go
@@ -29,9 +29,10 @@ func setMaxOpenFiles() error {
 	if err != nil {
 		return err
 	}
-	// Safe limit to be set on all platforms.
-	rLimit.Max = 4000
-	rLimit.Cur = 4000
+	// Set the current limit to Max, it is usually around 4096.
+	// TO increate this limit further user has to manually edit
+	// `/etc/security/limits.conf`
+	rLimit.Cur = rLimit.Max
 	err = syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
 	if err != nil {
 		return err

